license-maven-plugin/src/main/java/com/mycila/maven/plugin/license/AbstractLicenseMojo.java

```diff
@@ -94,9 +99,12 @@ public abstract class AbstractLicenseMojo extends AbstractMojo {
    * This is named `defaultBaseDirectory` as it will be used as the default
    * value for the base directory. This default value can be overridden
    * in each LicenseSet by setting {@link LicenseSet#basedir}.
+   *
+   * @deprecated use {@link WorkSpace#basedir}
    */
+  @Deprecated
   @Parameter(property = "license.basedir", defaultValue = "${project.basedir}", alias = "basedir", required = true)
-  public File defaultBasedir;
+  public File legacyDefaultBasedir;
```

Comment:
I would make it optional and null by default - in order to catch users using it explicitely.
`@Parameter(property = "license.basedir", alias = "basedir")`